### PR TITLE
Use AND (&&) operator to negate not nullish alternates

### PR DESCRIPTION
**Pull request by luisadame**
This PR adds support for transforming conditional expressions with alternates
that are not nullish (null || undefined). Eg:

```js
{propA ? 'Hello' : 'Bye'}
```

into a a pair of logical expressions, the second expression negating the test
in order to render the alternate

```js
{propA && 'Hello'}
{!propA && 'Bye'}
```

The implementation takes into account tests that are expressions and are not
simply identifiers which they look perfectly by just negating them.

So in cases where the test is like this:

```js
{1 + 2 === 3 ? 'Hello' : 'Bye'}
```

the test expression will be wrapped in parenthesis to negate the whole expression
instead introducing a runtime error. With a naive implementation the previous
example would be

```js
{!1 + 2 === 3 ? 'Hello' : 'Bye'}
```

This would be evaluated as the negation of the literal `1`, that would be `false`,
plus `2` equals `3`? And the answer would be `false` because the addition operation
will cast `false` to a number which evaluates to `0` therefore `0 + 2 === 3` is false

So, we could go "smart" and try to negate the expression in the operator level
by swapping `===` to `!==` but I think that doing that with every possible operator
would be overengineering this rule.

I think the case of having long test expressions is a problem of the developer
and if after the rule fixes the conditional the test expression ends up being
a big chunk of duplicated code the developer should extract the test to a variable
or function.

In summary, the example above:
```js
{1 + 2 === 3 ? 'Hello' : 'Bye'}
```
would be transformed into this
```js
{1 + 2 === 3 && 'Hello'}
{!(1 + 2 === 3) && 'Bye'}
```

The implementation to achieve this can be discussed here though.

As a second thought, I think consistency is great but sometimes standards are very
strict. In my opinion conditional expressions whose alternates are not nullish
should use ternary operators, as I think they look cleaner, less confusing and less
error prone, as we have to think how to negate the test, and perhaps we fail on doing
that.

So, should we add an option to indicate that when preferring AND operators if 
we find a conditional expressions that has a not nullish alternate should use
a ternary operator?

### CODE REVIEW

**Comment by luisadame** [comment]
Also, should be `not` or `non` nullish?

**Comment by luisadame** [comment]
# [Codecov](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez) Report
> Merging [#1](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez) (cf2e2be) into [main](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/commit/9419f6d4b501fbf028bbf528ca2d236558062a25?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez) (9419f6d) will **not change** coverage.
> The diff coverage is `100.00%`.

[Impacted file tree graph](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez)

```diff
@@            Coverage Diff            @@
##              main        #1   +/-   ##
=========================================
  Coverage   100.00%   100.00%           
=========================================
  Files            1         1           
  Lines          149       192   +43     
=========================================
+ Hits           149       192   +43     
```


| [Impacted Files](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez) | Coverage Δ | |
|---|---|---|
| [lib/rules/jsx-conditional.js](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez#diff-bGliL3J1bGVzL2pzeC1jb25kaXRpb25hbC5qcw==) | `100.00% <100.00%> (ø)` | |

------

[Continue to review full report at Codecov](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=footer&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez). Last update [9419f6d...cf2e2be](https://codecov.io/gh/luisadame/eslint-plugin-jsx-conditional/pull/1?src=pr&el=lastupdated&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments?utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Luis+Adame+Rodr%C3%ADguez).
